cmd/kubelet/app/server.go

```diff
@@ -1280,8 +1280,11 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	rl := make(v1.ResourceList)
 	for k, v := range m {
 		switch v1.ResourceName(k) {
-		// CPU, memory, local storage, and PID resources are supported.
-		case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs:
+		// CPU, memory, local storage, swap, and PID resources are supported.
+		case v1.ResourceCPU, v1.ResourceMemory, v1.ResourceEphemeralStorage, pidlimit.PIDs, v1.ResourceSwap:
+			if v1.ResourceName(k) == v1.ResourceSwap && !utilfeature.DefaultFeatureGate.Enabled(features.NodeSwap) {
+				continue
```

Comment:
setting a command line option that is not feature enabled should error, not get ignored silently